scripts/validate-front-matter/script.py

```diff
@@ -0,0 +1,180 @@
+NO_LINK_TITLE         = 'NO_LINK_TITLE'
+NO_OWNER              = 'NO_OWNER'
+NO_TITLE              = 'NO_TITLE'
+NO_TRAILING_NEWLINE   = 'NO_TRAILING_NEWLINE'
+NO_USER_QUESTIONS     = 'NO_USER_QUESTIONS'
+NO_WEIGHT             = 'NO_WEIGHT'
+REVIEW_TOO_LONG_AGO   = 'REVIEW_TOO_LONG_AGO'
+SHORT_DESCRIPTION     = 'SHORT_DESCRIPTION'
+
+def dump_result(rdict):
+    for key in rdict:
+        if len(rdict[key]) > 0:
+            print(f'\n{key}')
+            for item in sorted(rdict[key]):
+                print(f'  - {item}')
+
+
+def get_front_matter(source_text):
+    """
+    Tries to find front matter in the beginning of the document and
+    then returns a dict or raises an exception.
+    """
+    matches = list(re.finditer(r"(---)\n", source_text))
+    if len(matches) >= 2:
+        front_matter_start = matches[0].start(1)
+        front_matter_end = matches[1].start(1)
+        data = load(source_text[(front_matter_start + 3):front_matter_end], Loader=Loader)
+        return data
+    return None
+
+
+def main():
+    # Result dict has a key for each rule to check, where the value is a list of pages
+    result = {
+        INVALID_LAST_REVIEW_DATE: set(),
+        INVALID_OWNER: set(),
+        LONG_DESCRIPTION: set(),
+        LONG_LINK_TITLE: set(),
+        LONG_TITLE: set(),
+        LONG_USER_QUESTIONS: set(),
+        NO_DESCRIPTION: set(),
+        NO_FRONT_MATTER_FOUND: set(),
+        NO_LAST_REVIEW_DATE: set(),
+        NO_LINK_TITLE: set(),
+        NO_OWNER: set(),
+        NO_TITLE: set(),
+        NO_TRAILING_NEWLINE: set(),
+        NO_USER_QUESTIONS: set(),
+        NO_WEIGHT: set(),
+        REVIEW_TOO_LONG_AGO: set(),
+        SHORT_DESCRIPTION: set(),
+    }
+
+    # Iterate through pages
+    fpaths = []
+    for root, _, files in os.walk(path):
+        for file in files:
+            if not file.endswith('.md'):
+                continue
+            
+            fpaths.append(os.path.join(root, file))
+
+    top_attributes = set()
+
+    for fpath in fpaths:
+        fm = {}
+        with open(fpath, 'r') as input:
+            content = input.read()
+
+            if not content.endswith('\n'):
+                result[NO_TRAILING_NEWLINE].add(fpath)
+                # Can't parse reliably, so skipping further checks
+                continue
+
+            try:
+                fm = get_front_matter(content)
+            except Exception as e:
+                print(f'ERROR: page {fpath} has no front matter')
+        
+        if fm is None:
+            result[NO_FRONT_MATTER_FOUND].add(fpath)
+            continue
+        
+        # Collect all front matter keys
+        for key in fm:
+            top_attributes.add(key)
+
+        # Evaluate linkTitle
+        if 'linkTitle' in fm:
+            if len(fm['linkTitle']) > 40:
+                result[LONG_LINK_TITLE].add(fpath)
+        
+        # Evaluate title
+        if 'title' in fm:
+            if len(fm['title']) > 40 and not 'linkTitle' in fm:
```

Comment:
Why is the `linkTitle` checked here as well as line 137?

Reply:
This way linkTitle is required if title is longer than 40 characters.